### PR TITLE
fix: bail out of retry logic in swr if status 400, 403 or 404

### PR DESCRIPTION
**Pull request by hirbod**
# Why

Our Feed and data is cached. If we ban profiles, they might still be in the Feed cache. This causes hundreds of retries by useSWRs defaults, spamming our API and eating up client resources.

# How

While I think we should adapt a global SWRConfig strategy (even though we have exponential backoff baked in), for now I've gone with the route to only adjust this single useSWR. There is no need to retry to grab a profile with a 404 or 403 forever or hundreds of times. Therefore I have added this exception, until we decide how to proceed globally.

In my findings, a single banned profile was causing 80 XHR requests in less than 5 minutes on the home feed. This should fix this.

If you guys agree, we could ignore my PR and just add the exception to swr-provider.tsx and swr-prodiver.web.tsx and generally ignore retries for 403

# Test Plan

### CODE REVIEW

File: packages/app/hooks/api-hooks.ts [hunk 1]
```diff
@@ -116,7 +116,18 @@ export const useUserProfile = ({ address }: { address?: string | null }) => {
   const queryKey = address ? USER_PROFILE_KEY + address : null;
   const { data, error, isLoading } = useSWR<{
     data?: UserProfile;
-  }>(queryKey, fetcher);
+  }>(queryKey, fetcher, {
+    onErrorRetry: (error, key, _config, revalidate, { retryCount }) => {
+      // Never retry on 404 or 404.
```
Comment (hirbod) [inline packages/app/hooks/api-hooks.ts]:
```suggestion
      // Never retry on 403 or 404.
```

**Comment by intergalacticspacehighway** [comment]
> If you guys agree, we could ignore my PR and just add the exception to swr-provider.tsx and swr-prodiver.web.tsx and generally ignore retries for 403

yeah let's do it!

**Comment by alantoa** [comment]
Agreed! 💯

**Comment by hirbod** [comment]
@alantoa @intergalacticspacehighway you can review again please. Should be resolved now

**Review by alantoa — APPROVED** [review]
LGTM 💯

**Review by intergalacticspacehighway — APPROVED** [review]
LGTM ❇️

**Pull request merged**